PocketKit/Sources/PocketKit/Article/Presenters/ImageComponentPresenter.swift

```diff
@@ -68,6 +68,31 @@ class ImageComponentPresenter: ArticleComponentPresenter {
         return base
     }
     
+    var image: ImageComponentCell.ImageSpec? {
+        return imageCacheURL(for: component.source).flatMap {
+            ImageComponentCell.ImageSpec(
+                source: $0,
+                size: CGSize(
+                    width: lastAvailableWidth,
+                    height: .greatestFiniteMagnitude
+                )
+            )
+        }
+    }
+    
+    var shouldHideCaption: Bool {
+        return caption == nil || caption?.string.trimmingCharacters(in: .whitespaces).isEmpty == true
+    }
+    
+    func imageViewBackgroundColor(imageSize: CGSize) -> UIColor {
+        if let idealWidth = image?.size.width,
```

Comment:
I'm wondering what happens here if `image?.size.width` happens to be `nil` but `shouldHideCaption` is still true. We'd still want a `.clear` background color but I think in this (very edgey) case, we'd end up with a grey background.

Reply:
updated so that if no image, we return `.clear` background color